Library/Formula/node.rb

```diff
@@ -73,6 +73,10 @@ def install
       cd buildpath/"npm_install" do
         system "./configure", "--prefix=#{libexec}/npm"
         system "make", "install"
+        # `package.json` has relative paths to the npm_install directory.
+        # This copies back over the vanilla `package.json` that is expected.
+        # https://github.com/Homebrew/homebrew/issues/46131#issuecomment-157845008
+        cp buildpath/"npm_install/package.json", libexec/"npm/lib/node_modules/npm"
```

Comment:
My concern with this: what if new things are introduced into this file by newer versions of `npm`? Would it not be possible to `inreplace` the relative paths to absolute paths instead?